react-app/src/components/ActionForm/index.tsx

```diff
@@ -174,6 +174,8 @@ export const ActionForm = <Schema extends SchemaWithEnforcableProps>({
         pathnameToDisplayOn: formOrigins.pathname,
       });
 
+      // Prevent stale data from displaying on formOrigins page
+      await queryClient.invalidateQueries({ queryKey: ["record"] });
```

Comment:
Huh, I wonder why this doesn't work on line 146. Can we avoid the duplicate logic?

Reply:
I was wondering the same thing. It _does_ work on localhost, but it doesn't work when deployed. Seems to be just enough milliseconds to create a race condition. The await before redirecting seems to resolve that, and I agree the duplicate line feels clunky.

Reply:
@asharonbaltazar Also, thank you for your great refactor work on these files! It made tracking down this issue way easier.